packages/app/contexts/TokenListContext.tsx

```diff
@@ -1,199 +1,206 @@
+  1: [
+    "https://tokens.1inch.eth.link/",
+    "https://files.cow.fi/tokens/CoinGecko.json",
+    "https://files.cow.fi/tokens/CowSwap.json",
+  ],
+  100: [
+    "https://tokens.honeyswap.org/",
+    "https://files.cow.fi/tokens/CowSwap.json",
+  ],
+};
+
+const TokenListContext = createContext<{
+  tokenList: TokenFromTokenlist[];
+  tokenListWithBalances?: TokenWithBalance[];
+  getTokenLogoURL: (tokenAddress: string) => string;
+  getTokenFromList: (tokenAddress: string) => TokenFromTokenlist | false;
+}>({
+  tokenList: DEFAULT_TOKEN_LIST_BY_CHAIN[ChainId.GNOSIS],
+  getTokenLogoURL: (tokenAddress: string) => "#",
+  getTokenFromList: (tokenAddress: string) => false,
+});
+
+const mergeTokenlists = (
+  tokenList: TokenFromTokenlist[],
+  newTokenlist: TokenFromTokenlist[]
+) => {
+  const addresses = new Set(
+    tokenList.map((token) => token.address.toLowerCase())
+  );
+  const mergedLists = [
+    ...tokenList,
+    ...newTokenlist.filter(
+      (token) => !addresses.has(token.address.toLowerCase())
+    ),
+  ];
+
+  return mergedLists;
+};
+
+export const TokenListProvider = ({ children }: PropsWithChildren) => {
+  const [tokenList, setTokenList] = useState<TokenFromTokenlist[]>(
+    defaultGnosisTokenlist
+  );
+  const [tokenListWithBalances, setTokenListWithBalances] =
+    useState<TokenWithBalance[]>();
+  const { chain } = useNetwork();
+  const { address } = useAccount();
+
+  const chainId = chain?.id ?? ChainId.GNOSIS;
+
+  const defaultTokenList = chain
+    ? DEFAULT_TOKEN_LIST_BY_CHAIN[chain.id]
+    : DEFAULT_TOKEN_LIST_BY_CHAIN[ChainId.GNOSIS];
+
+  const fetchTokenlistURLS = chain
+    ? TOKEN_LISTS_BY_CHAIN_URL[chain.id]
+    : TOKEN_LISTS_BY_CHAIN_URL[ChainId.GNOSIS];
+
+  const callArray = useMemo(() => {
+    const erc20Interface = new ethers.utils.Interface(Erc20Abi);
+
+    if (address && tokenList) {
+      return tokenList.map((token) => [
+        token.address,
+        erc20Interface.encodeFunctionData("balanceOf", [address]),
+      ]);
+    }
+  }, [address, tokenList]);
+
+  /**
+   * Once the tokenList is defined and we have an address
+   * connected, we read data from the blockchain using
+   * the Provider through the Multicall SC to batch fetch
+   * all the ERC-20 token balances for the connected wallet
+   * and the supported token list, then we desc. sort them
+   */
+  useEffect(() => {
+    const multicallInterface = new ethers.utils.Interface(MulticallAbi);
+    const provider = new ethers.providers.JsonRpcProvider(RPC_LIST[chainId]);
+
+    if (address) {
+      provider
+        .call({
+          to: MULTICALL_ADDRESS,
+          data: multicallInterface.encodeFunctionData("aggregate", [callArray]),
+        })
+        .then((response: any) => {
+          const callResult = multicallInterface.decodeFunctionResult(
+            "aggregate",
+            response
+          );
+          const resultados = callResult.returnData;
```

Comment:
Let's keep it in english

Reply:
Maybe we can call it `returnData``

Reply:
Oh lol I haven't checked that I forgot to rename some variables, thanks for catch that!